app/src/main/kotlin/com/wire/android/ui/home/conversations/details/GroupConversationDetailsViewModel.kt

```diff
@@ -60,21 +69,25 @@ class GroupConversationDetailsViewModel @Inject constructor(
     private val deleteTeamConversation: DeleteTeamConversationUseCase,
     private val removeMemberFromConversation: RemoveMemberFromConversationUseCase,
     private val observeIsSelfUserMember: ObserveIsSelfUserMemberUseCase,
+    private val updateConversationMutedStatus: UpdateConversationMutedStatusUseCase,
     savedStateHandle: SavedStateHandle,
     qualifiedIdMapper: QualifiedIdMapper
 ) : GroupConversationParticipantsViewModel(
     savedStateHandle,
     navigationManager,
     observeConversationMembers,
     qualifiedIdMapper
-) {
+), GroupConversationDetailsBottomSheetEventsHandler {
 
     override val maxNumberOfItems: Int get() = MAX_NUMBER_OF_PARTICIPANTS
 
     private val conversationId: QualifiedID = qualifiedIdMapper.fromStringToQualifiedID(
         savedStateHandle.get<String>(EXTRA_CONVERSATION_ID)!!
     )
 
+    var conversationSheetContent: ConversationSheetContent? by mutableStateOf(null)
```

Comment:
we agreed that the ViewModel has no knowledge of the UI , but we can leave it as it is for now, but TODO would be nice here

Reply:
honestly: i don't get the problem. 
All the actions on SheetState happen out of the ViewModel. Here we just collect all the necessary data into data class, so what the problem?